src/services/action-flow/action-flow-service.ts

```diff
@@ -0,0 +1,74 @@
+import { logger } from "../../util/logger";
+import { v4 as uuidv4 } from "uuid";
+import { FileService, fileService } from "../file-service";
+import { actionFlowApi } from "../../api/action-flow-api";
+import * as AdmZip from "adm-zip";
+import * as FormData from "form-data";
+import * as fs from "fs";
+
+class ActionFlowService {
+    private static readonly METADATA_FILE_NAME = "metadata.json";
+
+    public async exportActionFlows(packageId: string, metadataFilePath: string): Promise<void> {
+        const exportedActionFlowsData = await actionFlowApi.exportRawAssets(packageId);
+        const tmpZip: AdmZip = new AdmZip(exportedActionFlowsData);
+
+        const zip = new AdmZip();
+        tmpZip.getEntries().forEach(entry => {
```

Comment:
Why are we creating a new zip? Can you not use the tmpZip?

Reply:
Somehow adding a zip with entries using the `adm-zip` constructor differs from adding the zip entries using the `addFile` method

If I used the constructor to create the zip, and then send that zip to the server I was receiving the error `invalid entry size (expected 2187395080 but got 158 bytes)` therefore I had to use this kind of a hacky workaround.
I also spent some time looking at the code of `AdmZip` on why this is happening and noticed that they have a flag in case the object is created with another zip on the constructor and they use it in a lot of places, but it was hard to find something else or something more concrete.
However, the bug was for sure in the library as also creating the zip manually or with other tools worked as expected

Btw I tried to explain this issue a little bit also on the [commit message](https://github.com/celonis/content-cli/pull/190/commits/4a60c8bcb78beff4d4ac9a3b5cf82e345a97acdb)

Reply:
Thanks for explaining!